src/main/java/math/Fraction.java

```diff
@@ -170,8 +170,10 @@ public Fraction divide(Fraction fraction) throws ArithmeticException {
      * @return a fraction that represents the result
      */
     public Fraction multiply(Fraction fraction) {
-        // TODO (GHBJIL)
-        return null;
+        if(this.denominator == 0 || fraction == Fraction.ZERO){
```

Comment:
Note that the `denominator` of  a `Fraction` object can not be 0, this is ensured by the constructor. Moreover, you should use `fraction.equals(Fraction.ZERO)` instead of `fraction == Fraction.ZERO`. This requires the `equals()` method, but this job has been done by someone else.